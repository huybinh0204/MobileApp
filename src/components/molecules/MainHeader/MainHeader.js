import { useNavigation } from '@react-navigation/native';
import PropTypes from 'prop-types';
import React from 'react';
import { IconSvg } from '_components/atoms';
import { ICONS, NAVIGATION } from '_constants';
import { normalize } from '_utilities/screen';
import { ActionButton, Container, HeaderTitle } from './MainHeader.styles';

const MainHeader = ({ showBackButton, showMenuButton, title }) => {
  const { navigate, goBack } = useNavigation();

  return (
    <Container>
      {showBackButton && (
        <ActionButton testID="BackButton" onPress={goBack}>
          <IconSvg icon={ICONS.backIcon} width={normalize(25)} height={normalize(25)} />
        </ActionButton>
      )}
      {showMenuButton && (
        <ActionButton testID="MenuButton" onPress={() => navigate(NAVIGATION.settings.main)}>
          <IconSvg icon={ICONS.menu} width={normalize(30)} height={normalize(30)} />
        </ActionButton>
      )}
      {title ? (
        <HeaderTitle>{title}</HeaderTitle>
      ) : (
        <IconSvg icon={ICONS.logo} width={normalize(35)} height={normalize(35)} />
      )}
    </Container>
  );
};

MainHeader.defaultProps = {
  showBackButton: false,
  showMenuButton: true,
  title: null,
};

MainHeader.propTypes = {
  showBackButton: PropTypes.bool,
  showMenuButton: PropTypes.bool,
  title: PropTypes.string,
};

export default MainHeader;
