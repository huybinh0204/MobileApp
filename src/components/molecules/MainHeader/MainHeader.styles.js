import styled from '@emotion/native';
import { BaseText } from '_components/atoms';
import { normalize } from '_utilities/screen';

export const Container = styled.View({
  alignItems: 'center',
  justifyContent: 'center',
  height: normalize(60),
});

export const ActionButton = styled.TouchableOpacity(({ theme }) => ({
  position: 'absolute',
  left: theme.spacing.general,
  zIndex: 10,
}));

export const HeaderTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.mainContentSemiBold,
  color: theme.colors.beta900,
}));
