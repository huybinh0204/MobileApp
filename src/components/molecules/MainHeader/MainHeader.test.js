import { render } from '@testing-library/react-native';
import React from 'react';
import Theme from '_utilities/Theme';
import MainHeader from './MainHeader';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    navigate: jest.fn(),
  }),
  DefaultTheme: {},
  DarkTheme: {},
}));

describe('MainHeader', () => {
  let props;

  beforeEach(() => {
    props = {
      title: 'title',
    };
  });

  describe('structure', () => {
    it('should match the snapshot', () => {
      const { toJSON } = render(
        <Theme>
          <MainHeader {...props} />
        </Theme>
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
