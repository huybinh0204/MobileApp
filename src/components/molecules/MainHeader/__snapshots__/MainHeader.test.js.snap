// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader structure should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 108.5,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 36,
        "opacity": 1,
        "position": "absolute",
        "zIndex": 10,
      }
    }
    testID="MenuButton"
  >
    <SvgMock
      height={54.5}
      width={54.5}
    />
  </View>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#1F1D25",
          "fontFamily": "Poppins-SemiBold",
          "fontSize": 29,
          "lineHeight": 40,
        },
        undefined,
      ]
    }
  >
    title
  </Text>
</View>
`;
